File: tests/sessions_test.py
import numpy as np

from taflow import Sessions


def test_sessions_lifecycle_and_reset():
    state = Sessions()
    state.extend([True, False], [10.0, 12.0], [8.0, 7.0])
    assert state.compute()[1][-1] == 12.0
    assert len(state) == 2
    state.reset()
    assert state.value is None

File: crates/taflow-core/src/indicators/sessions_test.rs
use super::sessions::Sessions;

#[test]
fn session_boundaries_and_reset_are_consistent() {
    let mut state = Sessions::new();
    assert_eq!(state.append(true, 10.0, 8.0).session_high, 10.0);
    assert_eq!(state.append(false, 12.0, 7.0).session_low, 7.0);
    state.reset();
    assert!(state.value().is_none());
}
